### PR TITLE
chore: replace sync.Map with concurrent.HashTrieMap

### LINKED ISSUE #8851: Replace `containers.SyncMap` with `concurrent.HashTrieMap` where possible
Results from https://github.com/siderolabs/discovery-service/pull/60 looks promising in our monitoring so lets continue our move there. 

Depends on https://github.com/cosi-project/runtime/issues/455

### PR DESCRIPTION
**Pull request by DmitriyMV**
Also bump `cosi-project/runtime` to the v0.4.4

Closes #8851

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

File: internal/integration/api/reset.go [hunk 1]
```diff
@@ -139,7 +140,8 @@ func (suite *ResetSuite) TestResetWithSpecState() {
 
 	userDisksToWipe := xslices.Map(
 		xslices.Filter(disks.Messages[0].Disks, func(disk *storage.Disk) bool {
-			return !disk.SystemDisk
+			// The DVD-ROM check should be removed for something more reliable.
+			return !disk.SystemDisk && !disk.Readonly && !strings.Contains(disk.Model, "DVD-ROM")
```
Comment (smira) [inline internal/integration/api/reset.go:144]: we can add a specific field that it's a CD, plus it should report a small size, I can look into that

Comment (DmitriyMV) [reply internal/integration/api/reset.go:144]: @smira yeah, that would be good

**Comment by DmitriyMV** [comment]
Remove integration-misc label cause previous run [completed successfully](https://github.com/siderolabs/talos/actions/runs/9450302019/job/26032017316?pr=8880).

**Pull request merged**